boot/bootstate20.go

```diff
@@ -715,12 +728,6 @@ func genericInitramfsSelectSnap(bs bootState20, modeenv *Modeenv, rootfsDir stri
 		return nil, nil, fmt.Errorf("%s snap %q does not exist on ubuntu-data", typeString, file)
 	}
 
-	if err != nil && isTrySnapError(err) {
-		// just log that we had issues with the try snap and continue with
-		// using the normal snap
-		logger.Noticef("unable to process try %s snap: %v", typeString, err)
-		return curSnap, nil, errTrySnapFallback
-	}
```

Comment:
would have swapping the ifs here also solved the problem:

```
	if err == errTrySnapFallback {
		// this should not actually return, it should immediately reboot
		return nil, initramfsReboot()
	}

	// now validate the chosen kernel snap against the modeenv CurrentKernel's
	// setting
	if strutil.ListContains(modeenv.CurrentKernels, first.Filename()) {
		return first, nil
	}
```

Reply:
Maybe, but then the checks that happen in `genericInitramfsSelectSnap` for try status and whether there is or not a try snap would not happen. And it makes sense that it returns in that case (no try snap but status is trying).

Reply:
I'm confused because afaict your change there only affect the case where status is default status, or I'm reading the code wrong

Reply:
That's true, but actually I think that is a bug. Status should be returned by revisions() even if there is no try snap, as otherwise
```
	// then we are trying a snap update and there should be a try snap
	if trySnap == nil {
		// it is unexpected when there isn't one
		logger.Noticef("try-%[1]s snap is empty, but \"%[1]s_status\" is \"trying\"", typeString)
		return curSnap, nil, errTrySnapFallback
	}
```
would be dead code. I have pushed a commit to fix that now.